File: sample/python/humancat/cat.py
STATUS = ['normal', 'sleepy', 'hungry', 'angry']

class Cat(object):

    def __init__(self, name):
        self.name = name
        self.status = 'normal'

    def get_name(self):
        return self.name
    
    def set_name(self, name):
        self.name = name

    def get_status(self):
        return self.status
    
    def set_status(self, status):
        if status not in STATUS:
            raise ValueError('Not allowed status')
        
        self.status = status

    def meow(self):
        return 'Meow~'
    

File: build_sample/go/module/humancat/cat.go
package main

type Cat struct {
	name   string
	status string
}

func (c Cat) get_name() string {
	return c.name
}

func (c Cat) set_name(name string) {
	c.name = name
}

func (c Cat) get_status() string {
	return c.status
}

func (c Cat) set_status(status string) {
	c.status = status
}

func (c Cat) meow() string {
	return "Meow"
}
